app/selections.ts

```diff
@@ -116,6 +116,10 @@ function datasetFromDatasetStore (historyDataset: DatasetStore): Dataset {
  *
  */
 
+export function selectLogLatestVersion (state: Store): VersionInfo | undefined {
+  return selectHasHistory(state) ? selectLog(state)[0] : undefined
+}
```

Comment:
don't  use the ternary operator to return undefined

```suggestion
export function selectLogLatestVersion (state: Store): VersionInfo? {
  if (selectHasHistory(state)) {
    return selectLog(state)[0]
  }
}
```

also, does the optional typescript operator work on function signatures? I'd love to use it as written  above if so

Reply:
i wonder how many times you are going to tell me not to use the ternary operator w/ undefined before i internalize it 🤦‍♀️

Reply:
also sadly no ? for optional returns, but if you want to read about the discussion surrounding it: https://github.com/microsoft/TypeScript/issues/7426